src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -363,6 +404,107 @@ private VariantContext removeNonRefAlleles(final VariantContext vc) {
         }
     }
 
+    /**
+     * Re-genotype (and re-annotate) a combined genomic VC
+     * @return a new VariantContext or null if the site turned monomorphic and we don't want such sites
+     */
+    private VariantContext regenotypeSomaticVC(final VariantContext originalVC, final ReferenceContext ref, final FeatureContext features, boolean includeNonVariants) {
+        Utils.nonNull(originalVC);
+
+        final VariantContext result;
+        if ( originalVC.isVariant()  && originalVC.getAttributeAsInt(VCFConstants.DEPTH_KEY,0) > 0 ) {
+            result = callSomaticGenotypes(originalVC);
+        } else if (includeNonVariants) {
+            result = originalVC;
+        }
+        else {
+            result = null;
+        }
+        return result;
+    }
+
+
+    /**
+     * Drop low quality alleles and call genotypes
+     * CombineGVCFs will convert calls to no-call (of varying ploidy, as is the case in somatic)
+     *
+     * @param vc input VariantContext with no-called genotypes
+     * @return a VC with called genotypes and low quality alleles removed, may be null
+     */
+    private VariantContext callSomaticGenotypes(final VariantContext vc) {
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
```

Comment:
This can wait to be declared until after the `for` loop in the same line where it's ready to `make()`.

Reply:
Actually, I see it used far below, but that distance is so great I would try to make the uses more local, maybe by using two different builders.

Reply:
I'd rather stick with one builder since this method is going to get called for every single variant and I don't want to waste time setting up another builder.  Unless it's not as slow as I think it is?